lib/scala/downloader/src/main/java/org/enso/downloader/http/PathProgressBodyHandler.java

```diff
@@ -110,7 +125,7 @@ public void onComplete() {
 
     @Override
     public CompletionStage<Path> getBody() {
-      return CompletableFuture.completedFuture(destination);
+      return result;
```

Comment:
This was the gist of the problem. This handler claimed that the download was finished (returned `CompletableFuture.completedFuture` almost immediately.